hbase-server/src/main/java/org/apache/hadoop/hbase/procedure2/store/region/RegionFlusherAndCompactor.java

```diff
@@ -207,8 +212,14 @@ private void flushLoop() {
   }
 
   private boolean shouldFlush(long changes) {
-    return region.getMemStoreHeapSize() + region.getMemStoreOffHeapSize() >= flushSize ||
+    boolean flush = region.getMemStoreHeapSize() + region.getMemStoreOffHeapSize() >= flushSize ||
       changes > flushPerChanges;
+    if (flush && LOG.isTraceEnabled()) {
+      LOG.trace("shouldFlush memStoreSize={}, flushSize={}, changes={}, flushPerChanges={}",
+        region.getMemStoreHeapSize() + region.getMemStoreOffHeapSize(), flushSize, changes,
```

Comment:
These values could change between calls, so I think we'd better store them in a local variable and then log them.